File: python/problem_6.py
square = lambda x: x**2

def sum_of_squares(N):
    return sum(map(square, range(1, N+1)))

def square_of_sum(N):
    return square(sum(range(1, N+1)))

assert square_of_sum(10) - sum_of_squares(10) == 2640
print(square_of_sum(100) - sum_of_squares(100))

File: c/problem_6.c
#include<stdio.h>

long sum_of_squares(long N) {
    long sum = 0;
    for(int i=1; i<=N; i++)
        sum += i*i;
    return sum;
}

long square_of_sum(long N) {
    long sum = 0;
    for(int i=1; i<=N; i++)
        sum += i;
    return sum*sum;
}

int main() {
    printf("%ld\n", square_of_sum(100) - sum_of_squares(100));
    return 0;
}
